Задача: Распределенная армия героев меча и магии.

piligrim-curious opened the pull request:
**Main**: Класс для демонстрации работы.

piligrim-curious committed:
Задача: Распределенная армия героев меча и магии.
Main: Класс для демонстрации работы.

iBerbasov commented on src/main/java/school/faang/distributed/army/of/heroes/of/sword/and/magic/Army.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.Collections;
+import java.util.List;
+
+@NoArgsConstructor
```
А зачем тут noargsConstructor ? Лучше добавлять аннотации по их необходимости

iBerbasov commented on src/main/java/school/faang/distributed/army/of/heroes/of/sword/and/magic/Unit.java:
```diff
@@ -0,0 +1,10 @@
+
+@AllArgsConstructor
+@Getter
+public class Unit {
```
Этот класс лучше сделать абстрактным

iBerbasov commented on src/main/java/school/faang/distributed/army/of/heroes/of/sword/and/magic/Unit.java:
```diff
@@ -0,0 +1,10 @@
+@AllArgsConstructor
+@Getter
+public class Unit {
+    protected int power;
```
А тут не нужен protected, лучше private - наследники этим полем не пользуются, им хватает и геттера

iBerbasov commented on src/main/java/school/faang/distributed/army/of/heroes/of/sword/and/magic/Squad.java:
```diff
@@ -0,0 +1,30 @@
+    private List<Unit> units = new ArrayList<>();
+
+    public void addUnit(Unit unit) {
+        if (units == null) {
```
Излишняя проверка, ведь поле units всегда инициализируется пустым листом
Если есть опасения, что он когда-то может стать нулом, то лучше отметить поле final

iBerbasov commented on src/main/java/school/faang/distributed/army/of/heroes/of/sword/and/magic/Squad.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    public int calculateSquadPower() {
+        if (units == null || units.isEmpty()) {
```
Тут тоже излишне, про нул уже написал, а про пустой - со стримом ничего не случиться от пустой коллекции

piligrim-curious committed:
Задача: Распределенная армия героев меча и магии.
Main: Класс для демонстрации работы.

Изменения в соответствии с замечаниями код-ревьюера.
